parity/snapshot_cmd.rs

```diff
@@ -257,20 +258,25 @@ impl SnapshotCommand {
 		let writer = PackedWriter::new(&file_path)
 			.map_err(|e| format!("Failed to open snapshot writer: {}", e))?;
 
-		let progress = Arc::new(Progress::new());
+		let progress = Arc::new(RwLock::new(Progress::new()));
 		let p = progress.clone();
 		let informant_handle = ::std::thread::spawn(move || {
 			::std::thread::sleep(Duration::from_secs(5));
-
 			let mut last_size = 0;
-			while !p.done() {
-				let cur_size = p.bytes();
-				if cur_size != last_size {
-					last_size = cur_size;
-					let bytes = ::informant::format_bytes(cur_size as usize);
-					info!("Snapshot: {} accounts (state), {} blocks, {} bytes", p.accounts(), p.blocks(), bytes);
+			loop {
```

Comment:
I wonder how much slower this loop would execute w.r.t. reading a `RwLock` instead of `AtomicUsize` in each iteration?

Reply:
Hopefully using a `RwLock` is faster, but I agree, a benchmark providing a hard number would be a good thing. The reason it's (hopefully) faster is there were several atomics getting written to with `SeqCst` ordering in several places, so having a single lock is hopefully faster.